spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mustache/MustacheAutoConfiguration.java

```diff
@@ -79,8 +81,31 @@ public void checkTemplateLocationExists() {
 	@Bean
 	@ConditionalOnMissingBean
 	public Mustache.Compiler mustacheCompiler(TemplateLoader mustacheTemplateLoader) {
-		return Mustache.compiler().withLoader(mustacheTemplateLoader)
-				.withCollector(collector());
+		PropertyMapper pm = PropertyMapper.get().alwaysApplyingWhenNonNull();
+		Mustache.Compiler compiler = Mustache.compiler();
+
+		compiler = pm.from(this.mustache.getDefaultValue())
+				.toInstance(compiler::defaultValue, compiler);
+		compiler = pm.from(this.mustache.getNullValue()).toInstance(compiler::nullValue,
+				compiler);
+
+		compiler = compiler.withLoader(mustacheTemplateLoader).withCollector(collector());
+
+		return compiler;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnProperty("spring.mustache.formatter.value")
+	public Mustache.Formatter mustacheFormatter() {
+		return (value) -> this.mustache.getFormatter().getValue();
```

Comment:
This ignores the input `value` and always returns the value of the `spring.mustache.formatter.value` property.